### PR TITLE
Used tempfile to determine whether a directory is writable.

### LINKED ISSUE #1734: Race condition for root environment detection
Periodically, when two conda processes are running at the same time, it is possible to see a race condition on determining whether the root environment is writable. Notice how the following produces two different configs from the same setup:

```
$ conda info & conda info 

Current conda install:

             platform : osx-64
        conda version : 3.18.3
  conda-build version : 1.18.0
       python version : 2.7.10.final.0
     requests version : 2.8.1
     root environment : /Users/pelson/miniconda  (read only)
  default environment : /Users/pelson/miniconda
     envs directories : /Users/pelson/.conda/envs
                        /Users/pelson/envs
                        /Users/pelson/miniconda/envs
        package cache : /Users/pelson/.conda/envs/.pkgs
                        /Users/pelson/envs/.pkgs
                        /Users/pelson/miniconda/pkgs
...

Current conda install:

             platform : osx-64
        conda version : 3.18.3
  conda-build version : 1.18.0
       python version : 2.7.10.final.0
     requests version : 2.8.1
     root environment : /Users/pelson/miniconda  (writable)
  default environment : /Users/pelson/miniconda
     envs directories : /Users/pelson/miniconda/envs
        package cache : /Users/pelson/miniconda/pkgs
...
```

The offending line is in https://github.com/conda/conda/blob/master/conda/config.py#L135-L143 and https://github.com/conda/conda/blob/master/conda/utils.py#L43-L54. My assumption is that the ```.conda-try-write``` is being removed by the other process, and the exception is being raised in the ```finally``` block.

### PR DESCRIPTION
**Pull request by pelson**
Closes #1734.

### CODE REVIEW

**Comment by ilanschnell** [comment]
Is this also cleaning up the test file?

**Comment by pelson** [comment]
> Is this also cleaning up the test file?

Yes. That is the reason for using the tempfile context.

File: conda/utils.py [hunk 1]
```diff
@@ -43,13 +45,10 @@ def can_open_all_files_in_prefix(prefix, files):
...
-        try:
-            with open(join(dir_path, '.conda-try-write'), mode='wb') as fo:
-                fo.write(b'This is a test file.\n')
-            return True
-        finally:
-            # XXX: If this raises an exception it will also return False
-            os.unlink(join(dir_path, '.conda-try-write'))
+        with tempfile.TemporaryFile(prefix='.conda-try-write',
+                                    dir=dir_path) as fo:
+            fo.write(b'This is a test file.\n')
```
Comment (pelson) [inline conda/utils.py]: This line is probably redundant, and could be replaced with ```pass```.

**Comment by groutr** [comment]
Is this ready to merge?

**Comment by pelson** [comment]
> Is this ready to merge?

From my point of view, yes.

Thanks!

**Pull request merged**